protocol/river-layout-unstable-v1.xml

```diff
@@ -0,0 +1,194 @@
+        safely be ignored by the client.
+
+        The serial of this event is used to identify subsequent events and
+        request as belonging to this layout request. Beware that the client
+        might need to handle multiple layout requests at the same time.
+      </description>
+      <arg name="view_amount" type="uint" summary="amount of views which are part of the layout"/>
+      <arg name="master_amount" type="uint" summary="proposed views in the master area"/>
+      <arg name="master_factor" type="fixed" summary="proposed master area factor"/>
+      <arg name="x" type="int" summary="x coordinate of useable area on output"/>
+      <arg name="y" type="int" summary="y coordinate of useable area on output"/>
+      <arg name="width" type="uint" summary="width of useable area on output"/>
+      <arg name="height" type="uint" summary="height of useable area on output"/>
+      <arg name="output" type="object" interface="wl_output"/>
+      <arg name="serial" type="uint" summary="serial of request"/>
+    </event>
+
+    <event name="advertise_view">
+      <description summary="make layout client aware of view">
+        This event is raised by the compositor after a request_layout event.
+        It contains additional information about one of the set of views for
+        which a layout has been requested.
+
+        It is guaranteed that every view in the layout will be advertised
+        exactly once, in an undefined order.
+
+        A client not interested in the additional information may ignore this
+        event.
+
+        The serial is the same as that of the layout request this event belongs
+        to.
+      </description>
+      <arg name="index" type="uint" summary="index of view in list"/>
+      <arg name="tag_set" type="uint" summary="bit field, active tags of view"/>
+      <arg name="app_id" type="string" summary="view app-id"/>
+      <arg name="serial" type="uint" summary="serial of layout request"/>
+    </event>
+
+    <event name="advertise_done">
+      <description summary="all views have been advertised">
+        The compositor raises this event when it has advertised all of the set
+        of views for which a layout has been requested.
+
+        A client not interested in the additional information may ignore this
+        event.
+
+        The serial is the same as that of the layout request this event belongs
+        to.
+      </description>
+      <arg name="serial" type="uint" summary="serial of layout request"/>
+    </event>
+
+    <request name="set_view_dimensions">
+      <description summary="propose view dimensions">
+        A client may use this to propose position and dimensions of one of the
+        set of views for which a layout has been requested.
+
+        It is not necessary to propose position and dimensions for every view.
+        If the client did not make a proposal of these for a view, they will be
+        determined by the compositor.
```

Comment:
What's your reasoning for allowing this? It seems to me it would be better to start off more strict and loosen up if needed.

Reply:
Sure.